### ISSUE TITLE
'thousands separator' / 'decimal point' is mixed up

### ISSUE THREAD

**Issue by fyahfox** [NewEconomyMovement/NEMAndroidApp#49]
v0.1.24.b1495

The thousands separator and decimal point is mixed up.
You are displaying a balance as 50.000,00 XEM, but when a user wants to send an amount of e.g. 10,25 the app says "invalid amount".

Please make sure to use "," as thousands separator and "." as decimal point in the english version. Other languages behave differently.

**Comment by am4n0w4r** [comment]
All number converting was moved to one place, should be consistent now, in 0.1.26

**Comment by fyahfox** [comment]
Did you also consider that different languages are using different separator chars?
E.g. in English it is 50,000.24 XEM, but in German it is 50.000,24 XEM.

**Comment by am4n0w4r** [comment]
Yes, all number processing consider device locale. Potential issues can arise if locale is overridden, please report if you find some. That is because Android officially does not support language overriding for apps, so there may be side effects with this.

**Comment by fyahfox** [comment]
ok, thank you. ill report back when i tested it

**Comment by fyahfox** [comment]
ok i think we have a special case here:
my android system language is german, so now the nemapp (v0.1.27.b1956) displays "Deutsch" in the language settings, which doesn't make sense, because there is no "Deutsch" translation for the app present.

also: it now uses "," as decimal point, which is correct for german,but since the app is NOT in german this is not how we can do it.

if the system language is not available, keep english as default and of course also use default english decimal point ".".

**Comment by fyahfox** [comment]
still weird on 0.1.28

language settings contains 3 languages total: 2 which i cant read and "Device default".

my device default is german, so why is "english" not in that list?

also: when i chose "device default", it still displays "Deutsch" when i go back to options...

**Comment by fyahfox** [comment]
To prevent accidents (sending bigger amounts than wanted), I suggest to always block the thousands separator (just like you block all chars besides numbers) and only allow decimal point for all amount textboxes where the user can enter numbers.

So when the app is in english: only "." will work and "," will be ignored.
When the app is e.g. in german: only "," will work and "." will be ignored.

**Comment by am4n0w4r** [comment]
Resolved in 0.1.30;

**Comment by fyahfox** [comment]
0.1.30

My phone is set to german language. App is in english of course (auto fallback, because there is no german language yet).
Yet for amounts the app only accepts "," and not ".", which would be correct for english.

**Comment by am4n0w4r** [comment]
This is a tricky issue.
Although I was able to change decimal separator to be the same as in app selected locale, and it's working,
this may lead to potential issues with different soft keyboard implementations when keyboard will show wrong separator (usually default to phone's locale because there is no way to set locale for keyboard) or no separators at all. This appears to be a known problem due to different keyboard implementations, and I experienced this on both stock LG keyboard (which doesn't show any separators) and another custom. Solution to this can be to use default stock android's [Google Keyboard](https://play.google.com/store/apps/details?id=com.google.android.inputmethod.latin), which seems to work fine with all cases.

Resolved in 0.1.31
